src/interfaces.d.ts

```diff
@@ -77,9 +79,28 @@ export interface HasValue {
 
 export interface ItemProperties<T = any> {
 	id: string;
+	index: number;
 	data: T;
 }
 
+/**
+ * @type SizeDetails
+ *
+ * Adds information about the size of a data set
+ *
+ * @property  dataLength  The number of items in the data set available through a slice operation
+ * @property  totalLength The total number of items in the data
```

Comment:
Well there's another API I will be adding that limits the data set itself so I'll reference that in totalLength when it's added.